real_decoder.go

```diff
@@ -121,6 +121,20 @@ func (rd *realDecoder) getArrayLength() (int, error) {
 	return tmp, nil
 }
 
+func (rd *realDecoder) getArrayLengthNoLimit() (int, error) {
```

Comment:
I’m not entirely sure why we would want to keep the two implementations around?

Reply:
I didn't want to change the behavior of everything else using the `getArrayLength` function. But if that's a worry I can just update  `getArrayLength` instead.

Reply:
Makes sense. Let’s see what the maintainers say. 👍